### ISSUE TITLE
Failing to create a .dot file in DotGroupCollaboration::writeGraph should terminate doxygen

### ISSUE THREAD

**Issue by cmorty** [doxygen/doxygen#6958]
I have some issues with dot not finding it's `.dot` file. While going through the code I found that if `QCString DotGroupCollaboration::writeGraph` fails to create the dot file, it ignores this and continues silently. 

https://github.com/doxygen/doxygen/blob/ac76b9a7a70b3c828c76442f5977937fcb87811a/src/dot.cpp#L4642-L4647

While I don't think that this causes my issue, not being able to create a .dot file should IMO terminate the process.

**Comment by albert-github** [comment]
I don't think that this is the cause of your error, at least a warning should be given in case it is not possible to write the file. Another discussion will be whether or not the process should be terminated or not, see also #642 and #6653.

Your problem probably is due to either:
- disk full
- no write permission to the write to the new to create dotfile.

would anyway be interesting to know what the name of the file is that it tries to open (any special characters / spaces / extra punctuations etc.).

**Comment by cmorty** [comment]
No, this is not related to my issue (I'll for that in a moment). It's rather a code review comment. ;)

**Comment by albert-github** [comment]
I don't understand the comment here (are this issue #6958 and #6959 not related? I'll await the bug report for the issue you just mentioned in the comment).

For this issue:
As written:
- it indeed would be better to have, at least, a warning about the failure.
To be able to successful see what the background of the issue is it would be nice to have an example so:
  - Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.
  - Please also specify the doxygen version used.

**Comment by cmorty** [comment]
As I said. Looking for the cause of the other issue I read through the code and my code-review-self said: "Wait, here's code where something can go wrong unnoticed!". - You can assign it to me and I'll try to look into it.
